app/code/Magento/ConfigurableProduct/Model/ConfigurableAttributeData.php

```diff
@@ -16,6 +18,22 @@
  */
 class ConfigurableAttributeData
 {
+    const XML_PATH_CONF_PRODUCT_ATTRIBUTES_SORT = 'catalog/frontend/product_attributes_sort_by';
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    protected $scopeConfig;
```

Comment:
Please change the protected property to private. According [Magento technical guideline](https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html)  we can't introduce new protected properties